master/buildbot/process/build.py

```diff
@@ -530,28 +529,38 @@ def startNextStep(self):
             return self.allStepsDone()
         self.executedSteps.append(s)
         self.currentStep = s
-        d = defer.maybeDeferred(s.startStep, self.conn)
-        d.addBoth(self._flushProperties)
-        d.addCallback(self._stepDone, s)
-        d.addErrback(self.buildException)
-        return None
+
+        # the following function returns a deferred, but we don't wait for it
+        self._start_next_step_impl(s)
+        return defer.succeed(None)
 
     @defer.inlineCallbacks
```

Comment:
I think we should keep this shortcut (perhaps skip the inlinecallback and results forwarding)

Reply:
Why? I think it makes code harder to read, because it's an indirection to a bit of logic that's almost impossible to get wrong. Also, we don't do this for other `self.master.data.updates` methods, so having a separate method introduces a little bit of confusion too.

Looking into history, previously it wasn't a single-line method, so it made sense to have it this way at the time. I don't think it's worthwhile now.

Reply:
I think the intention of flushProperties is important. This means that at those point, the properties are flushed into the db and available in the UI via events.

We shouldn't call this too often because there is performance impacts.

Reply:
I agree, but the same about availability of data to the UI can be said about other `setBuild*` functions. I guess if we call e.g. `setBuildStateString` too often we would see performance impact too :)